### PR TITLE
Fix #1176: restore code from #1150

### LINKED ISSUE #1151: [Cleanup] remove pool var related code from SpCodePresenter
Somehow I can not commit are removal of these methods (changing the sender to not call them worked and was already merged)

SpCodePresenter>>#browseImplementorOfPoolVar
SpCodePresenter>>#browsePoolVarNode:

TODO: remove these methods, deprecation not needed

### LINKED ISSUE #1176: strange: SpCodePresenter>>#doBrowseImplementors reverted to old version
We changed SpCodePresenter>>#doBrowseImplementors in #1150, but the current version in Pharo10 still have the (bad) code
that calls #browseImplementorOfPoolVar

### PR DESCRIPTION
**Pull request by guillep**
Fixes #1176, restores behavior from #1150.

Restore changes in doBrowseImplementors
Remove dead code related to pool variables

From this we realized that browsing implementors on variables is showing their users (wrong!).
Issue: https://github.com/pharo-spec/Spec/issues/1180

### CODE REVIEW

**Review by MarcusDenker — APPROVED** [review]

File: src/Spec2-Code/SpCodePresenter.class.st [hunk 1]
```diff
@@ -348,14 +307,12 @@ SpCodePresenter >> doBrowseHierarchy [
...
-	| result selectionIsPoolVar |
 
-	selectionIsPoolVar := self browseImplementorOfPoolVar.
-	selectionIsPoolVar ifTrue: [ ^ self ].
-	 
+	| result |
 	result := self
-		browseSelectedSelectorIfGlobal: [ :global | global browse ]
-		ifNotGlobal: [ :selector | self systemNavigation browseAllImplementorsOf: selector ].
+		          browseSelectedSelectorIfGlobal: [ :global | global browse ]
```
Comment (MarcusDenker) [inline src/Spec2-Code/SpCodePresenter.class.st:313]: this needs to be "global read browse", else we will browse the class of the global variable..

Comment (MarcusDenker) [reply src/Spec2-Code/SpCodePresenter.class.st:313]: (this was already done but lost, see https://github.com/pharo-spec/Spec/pull/1138 )

**Comment by guillep** [comment]
Ok, I'll update the PR.

File: src/Spec2-Code/SpCodePresenter.class.st (hunk 1 above)
Comment (guillep) [reply src/Spec2-Code/SpCodePresenter.class.st:313]: Well, actually it does not matter for this PR. the ifGlobal block is ignored inside `browseSelectedSelectorIfGlobal: globalBlock ifNotGlobal: nonGlobalBlock`. So it does not matter what we put into that block, it's never executed.

We should solve that as part of #1180.

**Comment by guillep** [comment]
Nope, I'll not :)

That change does not matter for this PR. The ifGlobal block is ignored inside `browseSelectedSelectorIfGlobal: globalBlock ifNotGlobal: nonGlobalBlock`. So it does not matter what we put into that block, it's never executed.

We should solve that as part of #1180.

**Pull request merged**